### PR TITLE
dev/core#1874 Fix fatal error by passing through the string formatTyp…

### PR DESCRIPTION
**Pull request by seamuslee001**
…e rather than the date format when adding datepicker fields from metadata

Overview
----------------------------------------
This aims to fix the fatal error reported by @demeritcowboy by ensuring that we pass along the string date format rather than the actual date format

Before
----------------------------------------
Fatal Error on new individual form

After
----------------------------------------
No fatal error and date displays the correct list of years

ping @totten @eileenmcnaughton @demeritcowboy

### CODE REVIEW

File: CRM/Core/Form.php [hunk 1]
```diff
@@ -1666,7 +1666,7 @@ public function addField($name, $props = [], $required = FALSE, $legacyDate = TR
         }
         else {
           $fieldSpec = CRM_Utils_Date::addDateMetadataToField($fieldSpec, $fieldSpec);
-          $attributes = ['format' => $fieldSpec['date_format']];
+          $attributes = ['format' => $fieldSpec['html']['formatType']];
```
Comment (totten) [inline CRM/Core/Form.php:1669]: So it looks like you've done an audit in this PR to fill-in the `formatType` for a long list of fields.

On general principle, it seems like there should be a fallback is in order (e.g. for any extensions which have DAOs that are non-conformant - for any fields that were missed by the audit)?

OTOH, when I grep for `date_format`, I'm having trouble spotting anything that supplies `date_format`. Maybe the old is based on some wonky assumption?

**Comment by demeritcowboy** [comment]
Did a little r-run and looks good. Am getting an E_NOTICE when saving a contact but I think that's a separate 5.28 problem not related `Warning: Invalid argument supplied for foreach() in CRM_Contact_BAO_Contact::create() (line 309 of .../CRM/Contact/BAO/Contact.php).`

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by yashodha** [comment]
@seamuslee001 works well for me as well.

**Comment by yashodha** [comment]
test this please

**Comment by demeritcowboy** [comment]
I can't recreate the E_WARNING and after the retests the PR test site where I saw it gone. So I'll assume it was caused by solar flares. So no objections from me on the PR.

File: CRM/Core/Form.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Core/Form.php:1669]: date_format is set in the line above - ie addDateMetadataToFields

```
        if (!isset($field['date_format'])) {
          $dateAttributes = CRM_Core_SelectValues::date($fieldMetaData['html']['formatType'], NULL, NULL, NULL, 'Input');
          $field['start_date_years'] = $dateAttributes['minYear'];
          $field['end_date_years'] = $dateAttributes['maxYear'];
          $field['date_format'] = $dateAttributes['format'];
          $field['is_datetime_field'] = TRUE;
          $field['time_format'] = $dateAttributes['time'];
          $field['smarty_view_format'] = $dateAttributes['smarty_view_format'];
        }
```

**Comment by yashodha** [comment]
@demeritcowboy 
the test failure looks unrelated. I agree we can merge this.

**Comment by monishdeb** [comment]
I have tested the patch on the cms build http://core-17836-52edd.test-1.civicrm.org:8001 and it works for me: 

core


and agree that test failure is unrelated. Hence merging this PR

**Pull request merged**